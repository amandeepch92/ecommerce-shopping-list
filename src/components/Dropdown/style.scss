@import 'assets/scss/variables.scss';
.dropdown{
    color:$white;
}