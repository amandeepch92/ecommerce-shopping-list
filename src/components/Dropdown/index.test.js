import { shallow } from 'enzyme';
import Dropdown from './index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
const { configure } = require("enzyme");
configure({ adapter: new Adapter() });


const props = {
    options: [
      {
        label: "Filter By Size",
        value: ""
      },
      {
        label: "XS",
        value: "XS"
      }
    ],
    selected: "S"
  };
  
  describe("Dropdown Component", () => {
    test("should render correctly", () => {
      const selectbox = shallow(<Dropdown {...props} />);
      expect(selectbox).toMatchSnapshot();
    });
  
    test("should call onChange", () => {
      const selectbox = shallow(<Dropdown {...props} />);
      const event = {
        target: { value: "X" }
      };
      selectbox.find("select").simulate("change", event);
      expect(selectbox.state().value).toBe("X");
    });
  });