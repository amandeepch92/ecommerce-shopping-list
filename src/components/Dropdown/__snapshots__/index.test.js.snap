// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render correctly 1`] = `
<div
  className="dropdown"
>
  <label
    className="filter-label"
    htmlFor="select-box"
  >
    Select
  </label>
  <select
    className="form-control"
    id="select-box"
    onChange={[Function]}
    value="S"
  >
    <option
      key=""
      value=""
    >
      Filter By Size
    </option>
    <option
      key="XS"
      value="XS"
    >
      XS
    </option>
  </select>
</div>
`;
